common/src/main/java/com/commercetools/payment/service/PaymentAdapterServiceImpl.java

```diff
@@ -84,18 +82,17 @@ public PaymentAdapterServiceImpl() {
     }
 
     @Override
-    public PaymentStatus getPaymentStatus(String ref) {
-        return null;
-    }
-
-    @Override
     public Optional<PaymentMethodInfo> getPaymentMethodInfo(String interfaceId, String method) {
-        return findPaymentServiceProvider(interfaceId).getAvailablePaymentMethods().stream().filter(m -> m.getMethod().equals(method)).findFirst();
+        return findPaymentServiceProvider(interfaceId)
+                .getAvailablePaymentMethods().stream()
+                .filter(m -> Objects.equals(m.getMethod(), method))
+                .findFirst();
     }
 
-    // TODO: what if no PSP is found?
     private PaymentServiceProvider findPaymentServiceProvider(String paymentInterfaceId) {
         return findAllPaymentServiceProviders().stream()
-                .filter(psp -> psp.getId().equals(paymentInterfaceId)).findFirst().get();
+                .filter(psp -> psp.getId().equals(paymentInterfaceId))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException(format("Payment interface %s not found", paymentInterfaceId)));
```

Comment:
It requires complete project re-factoring: we will _hardcode_ service providers implementations at compile time, instead of searching for them at runtime in classpath based on included dependencies.

Now `PaymentAdapterService`/`PaymentAdapterServiceImpl` are in _common_ library and independent from providers implementations, but in proposed approach we will need to couple _common_ library/interfaces with concrete PSP implementations.

 See [PaymentAdapterServiceImpl#PaymentAdapterServiceImpl](https://github.com/commercetools/commercetools-payment-integration-java/blob/bc677fc3a1f0528c26eb48e8816a387424e96a42/common/src/main/java/com/commercetools/payment/service/PaymentAdapterServiceImpl.java#L33-L39)